src/python/pants/backend/js/target_types.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from pants.engine.target import (
+    COMMON_TARGET_FIELDS,
+    Dependencies,
+    MultipleSourcesField,
+    SingleSourceField,
+    Target,
+    TargetFilesGenerator,
+)
+
+JS_FILE_EXTENSIONS = (".js",)
```

Comment:
This is the only extension we'd ever expect, right? Unlike C. Probably worth inlining this etc

Reply:
Oh good point on `.jsx`! Seems like `javascript_sources` to me from my limited React experience. So, okay, guess this is not premature generalization.